miden-tx/src/error.rs

```diff
@@ -165,3 +165,152 @@ impl fmt::Display for AuthenticationError {
+const ERR_FAUCET_NON_FUNGIBLE_ALREADY_EXISTS: u32 = 131108;
+const ERR_FAUCET_NON_FUNGIBLE_BURN_WRONG_TYPE: u32 = 131109;
+const ERR_FAUCET_NONEXISTING_TOKEN: u32 = 131110;
+const ERR_NOTE_INVALID_SENDER: u32 = 131111;
+const ERR_NOTE_INVALID_VAULT: u32 = 131112;
+const ERR_NOTE_INVALID_INPUTS: u32 = 131113;
+const ERR_NOTE_TOO_MANY_ASSETS: u32 = 131114;
+const ERR_VAULT_GET_BALANCE_WRONG_ASSET_TYPE: u32 = 131115;
+const ERR_VAULT_HAS_NON_FUNGIBLE_WRONG_ACCOUNT_TYPE: u32 = 131116;
+const ERR_VAULT_FUNGIBLE_MAX_AMOUNT_EXCEEDED: u32 = 131117;
+const ERR_VAULT_ADD_FUNGIBLE_ASSET_MISMATCH: u32 = 131118;
+const ERR_VAULT_NON_FUNGIBLE_ALREADY_EXISTS: u32 = 131119;
+const ERR_VAULT_FUNGIBLE_AMOUNT_UNDERFLOW: u32 = 131120;
+const ERR_VAULT_REMOVE_FUNGIBLE_ASSET_MISMATCH: u32 = 131121;
+const ERR_VAULT_NON_FUNGIBLE_MISSING_ASSET: u32 = 131122;
+const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_ONE_MUST_BE_ZERO: u32 = 131123;
+const ERR_ASSET_FORMAT_POSITION_TWO_MUST_BE_ZERO: u32 = 131124;
+const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_THREE_MUST_BE_FUNGIBLE_FAUCET_ID: u32 = 131125;
+const ERR_FUNGIBLE_ASSET_FORMAT_POSITION_ZERO_MUST_BE_WITHIN_LIMITS: u32 = 131126;
+const ERR_NON_FUNGIBLE_ASSET_POS_ONE_MUST_FUNGIBLE_FAUCET_ID: u32 = 131127;
+const ERR_NON_FUNGIBLE_ASSET_HIGH_BIT_SET: u32 = 131128;
+const ERR_FUNGIBLE_ASSET_MISMATCH: u32 = 131129;
+const ERR_NON_FUNGIBLE_ASSET_MISMATCH: u32 = 131130;
+const ERR_ACCOUNT_NONCE_INCR_MUST_BE_U32: u32 = 131131;
+const ERR_ACCOUNT_INSUFFICIENT_ONES: u32 = 131132;
+const ERR_ACCOUNT_SET_CODE_ACCOUNT_MUST_BE_UPDATABLE: u32 = 131133;
+const ERR_ACCOUNT_SEED_DIGEST_MISMATCH: u32 = 131134;
+const ERR_ACCOUNT_INVALID_POW: u32 = 131135;
+const ERR_NOTE_DATA_MISMATCH: u32 = 131136;
+const ERR_ASSET_NOT_FUNGIBLE_ID: u32 = 131137;
+const ERR_ASSET_INVALID_AMOUNT: u32 = 131138;
+const ERR_ASSET_NOT_NON_FUNGIBLE_ID: u32 = 131139;
+const ERR_INVALID_NOTE_TYPE: u32 = 131140;
+const ERR_NOTE_INVALID_TAG_PREFIX_FOR_TYPE: u32 = 131141;
+const ERR_NOTE_TAG_MUST_BE_U32: u32 = 131142;
+
+pub const KERNEL_ERRORS: [(u32, &str); 71] = [
+    (ERR_FAUCET_RESERVED_DATA_SLOT, "For faucets the slot FAUCET_STORAGE_DATA_SLOT is reserved and can not be used with set_account_item"),
+    (ERR_ACCT_MUST_BE_A_FAUCET, "Procedure can only be called from faucet accounts"),
+    (ERR_P2ID_WRONG_NUMBER_OF_INPUTS, "P2ID scripts expect exactly 1 note input"),
+    (ERR_P2ID_TARGET_ACCT_MISMATCH, "P2ID's target account address and transaction address do not match"),
+    (ERR_P2IDR_WRONG_NUMBER_OF_INPUTS, "P2IDR scripts expect exactly 2 note inputs"),
+    (ERR_P2IDR_RECLAIM_ACCT_IS_NOT_SENDER, "P2IDR's can only be reclaimed by the sender"),
+    (ERR_P2IDR_RECLAIM_HEIGHT_NOT_REACHED, "Transaction's reference block is lower than reclaim height. The P2IDR can not be reclaimed"),
+    (ERR_SWAP_WRONG_NUMBER_OF_INPUTS, "SWAP script expects exactly 9 note inputs"),
+    (ERR_SWAP_WRONG_NUMBER_OF_ASSETS, "SWAP script requires exactly 1 note asset"),
```

Comment:
not for this PR, but I'd group P2ID, SWAP etc. related errors separately from kernel errors.